app/views/docs/authentication-oauth.phtml

```diff
@@ -0,0 +1,190 @@
+</p>
+
+<ul class="phases clear" data-ui-phases>
+    <li>
+        <h3>Web</h3>
+        <div class="ide" data-lang="javascript" data-lang-label="Web SDK">
+        <pre class="line-numbers"><code class="prism language-javascript" data-prism>import { Client, Account } from "appwrite";
+
+const client = new Client()
+    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
+    .setProject('[PROJECT_ID]');               // Your project ID
+
+const account = new Account(client);
+
+// Go to OAuth provider login page
+account.createOAuth2Session('amazon');</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Flutter</h3>
+        <div class="ide" data-lang="dart" data-lang-label="Flutter SDK">
+            <pre class="line-numbers"><code class="prism language-dart" data-prism>import 'package:appwrite/appwrite.dart';
+
+final client = Client()
+    .setEndpoint('https://cloud.appwrite.io/v1') // Your API Endpoint
+    .setProject('[PROJECT_ID]');               // Your project ID
+
+final account = Account(client);
+
+// Go to OAuth provider login page
+await account.createOAuth2Session(provider: 'amazon');</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Android</h3>
+        <div class="ide" data-lang="kotlin" data-lang-label="Android SDK">
+            <pre class="line-numbers"><code class="prism language-kotlin" data-prism>import io.appwrite.Client
+import io.appwrite.services.Account
+
+val client = Client()
+    .setEndpoint("https://cloud.appwrite.io/v1") // Your API Endpoint
+    .setProject("[PROJECT_ID]")                // Your project ID
+
+val account = Account(client)
+
+// Go to OAuth provider login page
+account.createOAuth2Session(provider = "amazon")</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>Apple</h3>
+        <div class="ide" data-lang="swift" data-lang-label="Apple SDK">
+            <pre class="line-numbers"><code class="prism language-swift" data-prism>import Appwrite
+
+let client = Client()
+    .setEndpoint("https://cloud.appwrite.io/v1") // Your API Endpoint
+    .setProject("[PROJECT_ID]")                // Your project ID
+
+let account = Account(client)
+
+// Go to OAuth provider login page
+try await account.createOAuth2Session(provider: "amazon")</code></pre>
+        </div>
+    </li>
+    <li>
+        <h3>GraphQL</h3>
+        <div class="notice">
+            <p>OAuth 2 is not available through the GraphQL API. You can use the REST API or any Client SDK instead.</p>
+        </div>
+    </li>
+</ul>
+
+<p>
+    You'll be redirected to the OAuth 2 provider's login page to log in. 
+    Once complete, you'll be redirected to the redirect URL configured in your OAuth 2 provider.
+</p>
+
+<h2><a href="#profile" id="profile">OAuth 2 Profile</a></h2>
+<p>
+    [TODO: @steven fill the docs to access oauth account profile information]
```

Comment:
It's common to fetch information from an OAuth2 provider such as an avatar image. To do this you can use the access token from the OAuth2 provider and make API calls to the provider.

After creating an OAuth2 session, you can fetch the session with:

```javascript
import { Client, Account } from "appwrite";

const client = new Client();

const account = new Account(client);

account.getSession('current');
```

An OAuth2 [session](https://appwrite.io/docs/models/session) will have the following attributes set:

* provider: The OAuth2 Provider
* providerUid: User ID from the OAuth2 Provider
* provderAccessToken: Access token from the OAuth2 provider that can be used to make API calls to the OAuth2 Provider

Refer to the docs for the OAuth2 provider for how to make an API call with the access token.